Source/GOATracer/ViewModels/MainWindowViewModel.cs

```diff
@@ -0,0 +1,214 @@
+﻿using Avalonia.Interactivity;
+using CommunityToolkit.Mvvm.ComponentModel;
+using CommunityToolkit.Mvvm.Input;
+using GOATracer.Views;
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.ComponentModel;
+using System.Diagnostics;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace GOATracer.ViewModels
+{
+    /// <summary>
+    /// Represents the main view model for the application's main window, providing data binding and command handling
+    /// for the user interface. 
+    /// Uses the RayTracerModel as the underlying data model and interface to the model's data and logic.
+    /// </summary>
+    public partial class MainWindowViewModel : ViewModelBase, INotifyPropertyChanged
+    {
+        // RayTracer Model as an interface to the ViewModel
+        [ObservableProperty]
+        private RayTracerModel _rayTracerModel;
+        // Collection of all lights, hold the same items as in the RayTracerModel
+        public ObservableCollection<LightViewModel> Lights { get; } = new();
+        // Collection of enabled lights for rendering and displaying in UI
+        public ObservableCollection<LightViewModel> EnabledLights { get; } = new();
+        //selected Light in the combobox of the UI
+        [ObservableProperty]
+        private LightViewModel _selectedLight;
+        private int _lightCounter = 0;
+        // Commands for UI interactions, DeleteLightCommand is passed to LightViewModel instances
+        public IRelayCommand<string> Command { get; }
+        public IRelayCommand<string> DeleteLightCommand { get; }
+
+
+        public MainWindowViewModel()
+        {
+            // Initialize RayTracerModel with default camera settings
+            _rayTracerModel = new RayTracerModel()
+            {
+                CameraPositionX = 1,
+                CameraPositionY = 1,
+                CameraPositionZ = 1,
+                CameraRotationX = 1,
+                CameraRotationY = 1,
+                CameraRotationZ = 1,
+                ImageWidth = 800,
+                ImageHeight = 600
+            };
+
+
+            // Initialize commands
+            Command = new RelayCommand<string>(OnCommand);
+            DeleteLightCommand = new RelayCommand<string>(DeleteLight);
+
+            AddNewLight();
+            SelectedLight = Lights.First();
+
+        }
+
+        /// <summary>
+        /// Adds a new light to the scene and updates the collection of enabled lights.
+        /// </summary>
+        private void AddNewLight()
+        {
+            _lightCounter++;
+            var newLightModel = new Light() { name = $"Light {_lightCounter}", isEnabled = true };
+            _rayTracerModel.Lights.Add(newLightModel);
```

Comment:
```suggestion
            RayTracerModel.Lights.Add(newLightModel);
```

Reference the property instead of the backing field.